integration/src/main/java/scripts/testscripts/EnumerateJobs.java

```diff
@@ -0,0 +1,198 @@
+    resourceList =
+        ResourceMaker.makeResources(
+            ownerReferencedGcpResourceApi,
+            ownerControlledGcpResourceApi,
+            getWorkspaceId(),
+            getDataRepoSnapshotId(),
+            getDataRepoInstanceName(),
+            RESOURCE_COUNT);
+
+    logger.info("Created {} resources", resourceList.size());
+    logger.info("Cleaning up {} resources", resourceList.size());
+
+    ResourceMaker.cleanupResources(resourceList, ownerControlledGcpResourceApi, getWorkspaceId());
+    logger.info("Cleaned up {} resources", resourceList.size());
+  }
+
+  @Override
+  public void doUserJourney(TestUserSpecification testUser, WorkspaceApi workspaceApi)
+      throws Exception {
+
+    // Case 1: fetch all
+    EnumerateJobsResult fetchall =
+        alpha1Api.enumerateJobs(getWorkspaceId(), null, null, null, null, null, null);
+    logResult("fetchall", fetchall);
+
+    // Case 2: fetch by pages
+    String pageToken = null;
+    for (int pageCount = 1; true; pageCount++) {
+      EnumerateJobsResult page =
+          alpha1Api.enumerateJobs(getWorkspaceId(), PAGE_SIZE, pageToken, null, null, null, null);
+      logResult("page " + pageCount, page);
+      pageToken = page.getPageToken();
+      if (page.getResults().size() == 0) {
+        break;
+      }
+    }
+
+    // Case 4: filter by resource type
+    EnumerateJobsResult snapshots =
+        alpha1Api.enumerateJobs(
+            getWorkspaceId(), null, null, ResourceType.DATA_REPO_SNAPSHOT, null, null, null);
+    logResult("snapshots", snapshots);
+
+    // Case 5: filter by stewardship type
+    EnumerateJobsResult controlled =
+        alpha1Api.enumerateJobs(
+            getWorkspaceId(), null, null, null, StewardshipType.CONTROLLED, null, null);
+    logResult("controlled", controlled);
+
+    // Case 6: filter by resource and stewardship
+    EnumerateJobsResult controlledBuckets =
+        alpha1Api.enumerateJobs(
+            getWorkspaceId(),
+            null,
+            null,
+            ResourceType.GCS_BUCKET,
+            StewardshipType.CONTROLLED,
+            null,
+            null);
+    logResult("controlledBuckets", controlledBuckets);
+
+    // Case 7: validate error on invalid pagination params
+    ApiException invalidPaginationException =
+        assertThrows(
+            ApiException.class,
+            () -> alpha1Api.enumerateJobs(getWorkspaceId(), -5, null, null, null, null, null));
+    assertThat(invalidPaginationException.getMessage(), containsString("Invalid pagination"));
+
+    invalidPaginationException =
+        assertThrows(
+            ApiException.class,
+            () ->
+                alpha1Api.enumerateJobs(getWorkspaceId(), 22, "junktoken", null, null, null, null));
+    assertThat(invalidPaginationException.getMessage(), containsString("Invalid page token"));
+  }
+
+  private void logResult(String tag, EnumerateJobsResult result) {
+    logger.info(
```

Comment:
I couldn't figure out a reliable way to check the exact job results, because the results depend on all of the flights that Stairway has run since it has been cleared out. That is, it would depend on the sequence of other tests. It also depends on the order of resource generation in the `ResourceMaker`.

As a separate ticket, I plan to add a `submitted_since` filter on the API. I think with that and maybe some output from `ResourceMaker` I could do more validation. Added PF-1281 and a TODO about enhancing the test for this.